### PR TITLE
Only include the required information in the metadata field

### PR DESCRIPTION
**Pull request by robertparkinson**
https://eaflood.atlassian.net/browse/WATER-4543

As part of the work to replace the legacy system NALD we need to generate return logs. This PR refactors the metadata field so that it only includes the required information from the points and purposes tables.

### CODE REVIEW

File: app/services/jobs/return-logs/fetch-return-logs.service.js [hunk 1]
```diff
@@ -115,7 +147,17 @@ async function _fetchReturnRequirements (isSummer, licenceReference) {
...
+      builder.select(['description',
+        'description',
+        'ngr1',
+        'ngr2',
+        'ngr3',
+        'ngr4'])
+    })
+    .withGraphFetched('returnRequirementPurposes.primaryPurpose')
+    .withGraphFetched('returnRequirementPurposes.secondaryPurpose')
+    .withGraphFetched('returnRequirementPurposes.purpose')
```
Comment (Cruikshanks) [inline app/services/jobs/return-logs/fetch-return-logs.service.js:160]: If you only need the legacy IDs and descriptions off these, then they should have `modifyGraph()` functions which specify this in the select.

Without it, Objection.js will do `SELECT *` pulling everything off the record, and for which I have a strict 🚫 rule.

File: test/services/jobs/return-logs/fetch-return-logs.service.test.js [hunk 2]
```diff
@@ -9,8 +9,10 @@ const { expect } = Code
 
 // Test helpers
 const LicenceHelper = require('../../../support/helpers/licence.helper.js')
+const PrimaryPurposeModel = require('../../../../app/models/primary-purpose.model.js')
+const SecondaryPurposeModel = require('../../../../app/models/secondary-purpose.model.js')
+const PurposeModel = require('../../../../app/models/purpose.model.js')
```
Comment (Cruikshanks) [inline test/services/jobs/return-logs/fetch-return-logs.service.test.js:14]: Just need to sort the ordering of these requires, including the existing ones.

File: test/services/jobs/return-logs/fetch-return-logs.service.test.js [hunk 3]
```diff
@@ -47,6 +49,10 @@ describe('Fetch return logs service', () => {
     it('should return one return log payload', async () => {
       const result = await FetchReturnLogsService.go(false, licence.licenceRef)
 
+      const primaryPurpose = await PrimaryPurposeModel.query().findById(returnRequirementPurpose.primaryPurposeId)
+      const secondaryPurpose = await SecondaryPurposeModel.query().findById(returnRequirementPurpose.secondaryPurposeId)
+      const tertiaryPurpose = await PurposeModel.query().findById(returnRequirementPurpose.purposeId)
+
```
Comment (Cruikshanks) [inline test/services/jobs/return-logs/fetch-return-logs.service.test.js:55]: These records are pre-seeded in the DB.

We should be able to use their helper `select()` functions to pick them, then assign those picked ones to your test return requirement purposes (just as done with `Region.select()`).

You can then use them again to assert the legacy IDs and descriptions are what you expected.

**Review by Cruikshanks — CHANGES_REQUESTED** [review]

File: app/services/jobs/return-logs/fetch-return-logs.service.js [hunk 4]
```diff
@@ -11,26 +11,29 @@ const ReturnVersionModel = require('../../../models/return-version.model.js')
...
 
-const endOfSummerCycle = new Date(new Date().getFullYear() + 1, summerEndMonth, summerEndDay)
-const endOfWinterAndAllYearCycle = new Date(new Date().getFullYear() + 1, allYearEndMonth, allYearEndDay)
+const endOfSummerCycle = new Date(new Date().getFullYear() + 1, SUMMER_END_MONTH, SUMMER_END_DAY)
+const endOfWinterAndAllYearCycle = new Date(new Date().getFullYear() + 1, ALL_YEAR_END_MONTH, ALL_YEAR_END_DAY)
 
 /**
  * Fetch all return requirements that need return logs created.
  *
+ * @param isSummer - are we running summer cycel or all year - boolean
```
Comment (Beckyrose200) [inline app/services/jobs/return-logs/fetch-return-logs.service.js:34]:
```suggestion
 * @param {boolean} isSummer - are we running summer cycle or all year
```

File: app/services/jobs/return-logs/fetch-return-logs.service.js [hunk 5]
```diff
@@ -11,26 +11,29 @@ const ReturnVersionModel = require('../../../models/return-version.model.js')
...
+ * @param licenceReference - if provided only do the return log for that licence reference - string
```
Comment (Beckyrose200) [inline app/services/jobs/return-logs/fetch-return-logs.service.js:35]:
```suggestion
 * @param {string} licenceReference - if provided only do the return log for that licence reference
```

**Review by Cruikshanks — APPROVED** [review]

**Pull request merged**